### PR TITLE
Feature/sinf 348 auth service

### PR DESCRIPTION
**Pull request by adrianmilne**
Will create a separate PR for my Postman collection. Tests still to be fleshed out - any pointers/suggestions on the rest gratefully received! 😄

### CODE REVIEW

File: src/main/java/uk/gov/crowncommercial/dsd/api/auth/model/account/Address.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import lombok.Data;
+
+/**
+ * CCS Address.
+ *
+ */
+@Data
+public class Address {
+
+  @JsonProperty("id")
```
Comment (tommyb82) [inline src/main/java/uk/gov/crowncommercial/dsd/api/auth/model/account/Address.java:13]: We could drop most of these explicit `@JsonProperty` annotations I think.. I was in two minds but got loads of them too.

File: src/main/java/uk/gov/crowncommercial/dsd/api/auth/logic/AccountResponseProcessor.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.crowncommercial.dsd.api.auth.converter.AccountConverter;
+import uk.gov.crowncommercial.dsd.api.auth.model.spree.account.Account;
+
+@Component
+public class AccountResponseProcessor implements Processor {
```
Comment (tommyb82) [inline src/main/java/uk/gov/crowncommercial/dsd/api/auth/logic/AccountResponseProcessor.java:11]: I think you could drop this explicit processor if you apply the `@Converter` annotation to [AccountResponse toAccountResponse](https://github.com/Crown-Commercial-Service/ccs-scale-api-tx-auth-service/pull/1/files#diff-3c1cbea97b70f6c8ba8159ad8eebd5e4ab9d891898d6f04edfb49b0cd520ceaaR18), then you should just be able to call `.convertBodyTo(AccountResponse.class)` - entirely optional though 👍

**Review by tommyb82 — APPROVED** [review]
Faultless as always and very neat 'n' tidy! Couple of very minor totally optional comments.  There's more we can/should do around exception mapping and retries of the Spree API call (e.g. on receipt of a `503` from the upstream service).. but.. all in good time 😎

File: src/main/java/uk/gov/crowncommercial/dsd/api/auth/logic/AccountResponseProcessor.java (hunk 2 above)
Comment (adrianmilne) [reply src/main/java/uk/gov/crowncommercial/dsd/api/auth/logic/AccountResponseProcessor.java:11]: Thanks - will look at including in a follow up PR as part of the tests

**Pull request merged**